0000-ilp-over-http.md

```diff
@@ -0,0 +1,59 @@
+---
+title: ILP Over HTTP
+draft: 1
+---
+
+# ILP Over HTTP
+> A bilateral communication protocol for server-to-server connections
+
+## Motivation
+
+Scaling Interledger infrastructure to handle large volumes of ILP packets requires horizontally scaling connectors. Using HTTP for bilateral communication enables service providers to leverage standard tools and services for hosting, load balancing, Distributed Denial of Service (DDoS) protection, and monitoring.
+
+## Overview
+
+In an ILP Over HTTP connection, both peers run HTTP servers with accessible HTTPS endpoints. When peering, the peers exchange their respective URLs, authentication tokens or TLS certificates, ILP addresses, and settlement-related details.
+
+Each ILP Prepare packet is sent as the body of an HTTP request to the peer's server endpoint. ILP Fulfill or Reject packets are returned as the body of the HTTP response.
+
+## Specification
+
+This is a minimal protocol built on HTTP. HTTP/2 is HIGHLY RECOMMENDED for performance reasons, although HTTP/1.1 MAY also be used. Implementations SHOULD support HTTP version negotiation via Application Protocol Negotiation (ALPN).
+
+### Authentication
+
+Peers MAY use any standard HTTP authentication mechanism to authenticate incoming requests. Bearer Tokens or TLS Client Certificates are RECOMMENDED.
+
+### Request
+
+```http
+POST /ilp HTTP/x.x
+Accept: application/ilp+octet-stream
+Content-Type: application/ilp+octet-stream
+Authorization: Bearer zxcljvoizuu09wqqpowipoalksdflksjdgxclvkjl0s909asdf
+Host: connector.example
+ILP-Destination: example.usd.connector.account
+
+< Body: Binary OER-Encoded ILP Prepare Packet >
+```
+
+**Field Details:**
+
+- **Path** - A connector MAY specify any HTTP path for their peer to send ILP packets to.
+- **ILP-Destination Header** - (Optional) It is RECOMMENDED to set this header to the destination ILP address from the ILP packet. This enables standard HTTP load balancers to route packets to particular connector instances without parsing the ILP packet.
```

Comment:
I'm not really sold on this. The load balancers in google cloud and AWS don't support features this advanced. It's possible you could do it with HAproxy or something but if it requires a lot of customization then you might as well write a load balancer that understands ILP packets (basically a connector at that point)

Reply:
You're right, I actually misread something in the Google Cloud Load Balancer and thought they did support this. Client IP-based session affinity should be fine.

Reply:
HAProxy definitely does support this (and I think nginx) so I wouldn't exclude it. 
I agree that we shouldn't make it mandatory. 

If you did the header + payload serialisation you'd have this header anyway so that could be a way for peers to expose it if they decide they need it.